tests/acceptance/features/bootstrap/SpacesTUSContext.php

```diff
@@ -0,0 +1,161 @@
+ * @copyright Copyright (c) 2022 Viktor Scharf v.scharf@owncloud.com
+ */
+
+use Behat\Behat\Context\Context;
+use Behat\Behat\Hook\Scope\BeforeScenarioScope;
+use GuzzleHttp\Exception\GuzzleException;
+use TestHelpers\HttpRequestHelper;
+use Behat\Gherkin\Node\TableNode;
+use TestHelpers\WebDavHelper;
+
+require_once 'bootstrap.php';
+
+/**
+ * Context for the TUS-specific steps using the Graph API
+ */
+class SpacesTUSContext implements Context {
+
+	/**
+	 * @var FeatureContext
+	 */
+	private FeatureContext $featureContext;
+
+	/**
+	 * @var TUSContext
+	 */
+	private TUSContext $tusContext;
+
+	/**
+	 * @var SpacesContext
+	 */
+	private SpacesContext $spacesContext;
+
+	/**
+	 * @var string
+	 */
+	private string $baseUrl;
+
+	/**
+	 * This will run before EVERY scenario.
+	 * It will set the properties for this object.
+	 *
+	 * @BeforeScenario
+	 *
+	 * @param BeforeScenarioScope $scope
+	 *
+	 * @return void
+	 */
+	public function before(BeforeScenarioScope $scope): void {
+		// Get the environment
+		$environment = $scope->getEnvironment();
+		// Get all the contexts you need in this context from here
+		$this->featureContext = $environment->getContext('FeatureContext');
+		$this->spacesContext = $environment->getContext('SpacesContext');
+		$this->tusContext = $environment->getContext('TUSContext');
+		$this->baseUrl = \trim($this->featureContext->getBaseUrl(), "/");
+	}
+
+	/**
+	 * @Given /^user "([^"]*)" has uploaded a file from "([^"]*)" to "([^"]*)" via TUS inside of the space "([^"]*)" using the WebDAV API$/
+	 *
+	 * @param string $user
+	 * @param string $source
+	 * @param string $destination
+	 * @param string $spaceName
+	 *
+	 * @return void
+	 *
+	 * @throws Exception
+	 * @throws GuzzleException
+	 */
+	public function userHasUploadedFileViaTusInSpace(string $user, string $source, string $destination, string $spaceName): void {
+		$this->userUploadsAFileViaTusInsideOfTheSpaceUsingTheWebdavApi($user, $source, $destination, $spaceName);
+		$this->featureContext->theHTTPStatusCodeShouldBe(200, "Expected response status code should be 204");
+	}
+
+	/**
+	 * @When /^user "([^"]*)" uploads a file from "([^"]*)" to "([^"]*)" via TUS inside of the space "([^"]*)" using the WebDAV API$/
+	 *
+	 * @param string $user
+	 * @param string $source
+	 * @param string $spaceName
+	 * @param string $destination
+	 * @param array|null $uploadMetadata
+	 *
+	 * @return void
+	 *
+	 * @throws Exception
+	 * @throws GuzzleException
+	 */
+	public function userUploadsAFileViaTusInsideOfTheSpaceUsingTheWebdavApi(string $user, string $source, string $destination, string $spaceName, ?array $uploadMetadata = null) {
+		$space = $this->spacesContext->getSpaceByName($user, $spaceName);
+		if (!WebDavHelper::$SPACE_ID_FROM_OCIS) {
+			WebDavHelper::$SPACE_ID_FROM_OCIS = $space['id'];
+		}
+		$this->tusContext->userUploadsUsingTusAFileTo($user, $source, $destination);
+		WebDavHelper::$SPACE_ID_FROM_OCIS = null;
```

Comment:
```suggestion
		WebDavHelper::$SPACE_ID_FROM_OCIS = '';
```

Reply:
THis step is no longer needed here handled from OC10